common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDBIterator.java

```diff
@@ -176,22 +183,33 @@ public boolean skip(long n) {
 
   @Override
   public synchronized void close() throws IOException {
-    db.notifyIteratorClosed(this);
+    db.notifyIteratorClosed(it);
     if (!closed) {
-      it.close();
-      closed = true;
-      next = null;
+      try {
+        it.close();
+        closed = true;
+        next = null;
```

Comment:
Keep both level db and rocks db part consistent w.r.t finally ? (move setting closed/next to finally ?)

Reply:
+1

Reply:
Thanks, done